### PR TITLE
Per 607 add json config service

### PR DESCRIPTION
**Pull request by pbelanger-avid**
This PR adds a service endpoint to the realsense2_camera node that allows updating the JSON configuration of the node at runtime. It allows consumers to monitor and modify all regular and "advanced" mode settings and change them while the camera is running. 

This functionality piggybacks off of the existing rs2::serializable_device API. 

The service accepts a string representation of a JSON object, and any fields present will be applied to the current device. 

The service always returns the current JSON configuration of the realsense device upon completion, so the success of a parameter application can be verified by comparing the commanded value to the actual value returned. 

The current configuration of the node can be obtained by passing an empty string to the service.

Note: this API was tested at one of our development sites but was since rebased off of the most recent AUTO-110 branch (including the major refactoring that was merged in from devel).

In particular, this new service exposes the adjustment of mean_intensity_setpoint for controlling the autoexposure setpoint of the camera.

### CODE REVIEW

**Review by jake5452 — APPROVED** [review]

**Review by jake5452 — APPROVED** [review]

**Comment by pbelanger-avid** [comment]
Apparently on Github I can't add myself as a reviewer but I approve of this PR.

File: realsense2_camera/include/realsense2_camera/realsense_node.h [hunk 1]
```diff
@@ -151,6 +152,7 @@ class RealSenseParamManager;
         static std::string getNamespaceStr();
         void getParameters();
         bool enableStreams(std_srvs::SetBool::Request  &req, std_srvs::SetBool::Response &res);
+        bool JsonConfigCallback(JsonConfig::RequestType &request, realsense2_camera::JsonConfig::ResponseType &response);
```
Comment (pbelanger-avid) [inline realsense2_camera/include/realsense2_camera/realsense_node.h]: Inconsistent use of the `realsense2_camera` namespace

File: realsense2_camera/scripts/rs2_json_config.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        print(result)
+
+    elif len(sys.argv) == 2:
+        ns = sys.argv[1]
+
+        result = json_config_client(ns, "")
+        print(result)
+
+    else:
+        print("usage: {} <camera ns> [new json config string] ")
```
Comment (ehollander-avidbots) [inline realsense2_camera/scripts/rs2_json_config.py]: Why the `{}` ? I don't see a `.format()` at the end of the string.

File: realsense2_camera/src/realsense_node.cpp [hunk 3]
```diff
@@ -263,6 +266,33 @@ bool RealSenseNode::enableStreams(std_srvs::SetBool::Request& req, std_srvs::Set
     return true;
 }
 
+bool RealSenseNode::JsonConfigCallback(JsonConfig::RequestType &request, JsonConfig::ResponseType &response)
+{
+    // obviously this only works if the device supports serialilzation in the first place...
+    if(not _dev.is<rs2::serializable_device>())
```
Comment (ehollander-avidbots) [inline realsense2_camera/src/realsense_node.cpp]: `not` ->`!`

File: realsense2_camera/include/realsense2_camera/realsense_node.h (hunk 1 above)
Comment (jake-avidbots) [reply realsense2_camera/include/realsense2_camera/realsense_node.h]: Nice catch will fix it!

File: realsense2_camera/scripts/rs2_json_config.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+    elif len(sys.argv) == 2:
+        ns = sys.argv[1]
+
+        result = json_config_client(ns, "")
+        print(result)
+
+    else:
+        print("Usage:")
+        print("{} <camera ns> to print all available json config parameters".format("rosrun realsense2_camera rs2_json_config.py"))
```
Comment (ehollander-avidbots) [inline realsense2_camera/scripts/rs2_json_config.py:67]: I'm confused as to why you didn't just write `print("rosrun realsense2_camera rs2_json_config.py <camera ns> to print all available json config parameters.")` but I guess this works.
Also, consider separating the command from the text describing what it does, either by putting them on a new line or adding some additional spaces between them (perhaps so both this "to" and the "to" from the line below start in the same column).

**Comment by pbelanger-avid** [comment]
I approve of this PR

**Pull request merged**